### DISCUSSION TITLE
Capabilities.os and iPad

### DISCUSSION THREAD

**Discussion by pol2095** [airsdk/Adobe-Runtime-Support#3424] [Q&A] [answered]
Hello,

what is returned by `Capabilities.os` for an iPad ?

Thanks

**Comment by ajwfrost** [comment]
Testing here with output of most of the `Capabilities` class, it's something like `iPadOS [os_version] [iPad_product_code]`
For the "product code" see https://gist.github.com/adamawolf/3048717

So e.g.
```
CPU arch = ARM
CPU size = 64
Lang = en
OS = iPadOS 17.6 iPad8,6
Supports 32bit = false
Supports 64bit = true
Version = lOS 51,1,1,2
HRes = 2048
VRes = 1536
Debugger = true
Manufacturer = Adobe macOS
Pixel aspect ratio = 1
Player type = Desktop
Screen DPI = 264
```

(although, this was when we run an iPad application on a macOS machine, hence the version of the OS is actually my macOS version, and the manufacturer code is different.. - I can get this test app run on an iPad tomorrow if you needed a proper result?! from memory though, it's essentially just like this..)

**Reply by pol2095** [reply]
Thanks, that's what I thought, if you can try on a real device...

**Reply by ajwfrost** [reply, marked as answer]
Screenshot from a 2022 iPad (10th generation)

![image](https://github.com/user-attachments/assets/21eae51e-aadb-49fd-bdf2-6c94495c432b)

**Reply by pol2095** [reply]
Thanks
